sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -194,6 +194,17 @@ case class UnknownPartitioning(numPartitions: Int) extends Partitioning {
   override def guarantees(other: Partitioning): Boolean = false
 }
 
+case class RandomPartitioning(numPartitions: Int) extends Partitioning {
```

Comment:
Argh, change my mind: going to stick with RoundRobin, since these names are actually describing the strategies used to distribute records. EvenlyDistributed sounds more like the name of a distribution than a partitioning anyways.